pkg/server/registry/images/tag.go

```diff
@@ -58,37 +59,35 @@ func (t *TagStrategy) ImageTag(ctx context.Context, namespace, imageName string,
 	if err != nil {
 		return nil, err
 	}
-
-	duplicateTag := make(map[string]bool)
-	for _, tag := range tags {
-		imageParsedTag, err := name.NewTag(tag, name.WithDefaultRegistry(""))
-		if err != nil {
-			return nil, err
-		}
-		duplicateTag[imageParsedTag.Name()] = true
+	set := sets.NewString()
+	for _, tag := range append(image.Tags, normalizeTags(tags)...) {
+		set.Insert(tag)
 	}
 
 	for _, img := range imageList.Items {
-		if img.Name == image.Name {
-			continue
-		}
-		for i, tag := range img.Tags {
-			imageParsedTag, err := name.NewTag(tag, name.WithDefaultRegistry(""))
-			if err == nil {
-				tag = imageParsedTag.Name()
-			} else {
-				logrus.Errorf("invalid tag [%s] found, ignoring: %v", tag, err)
-			}
-			if duplicateTag[tag] {
+		for i, tag := range normalizeTags(img.Tags) {
+			if set.Has(tag) && img.Name != image.Name {
 				img.Tags = append(img.Tags[:i], img.Tags[i+1:]...)
-				err := t.client.Update(ctx, &img)
+				err = t.client.Update(ctx, &img)
 				if err != nil {
 					return image, err
 				}
 			}
 		}
 	}
 
-	image.Tags = append(image.Tags, tags...)
+	image.Tags = set.List()
 	return image, t.client.Update(ctx, image)
 }
+
+func normalizeTags(tags []string) []string {
+	var result []string
+	for _, tag := range tags {
+		imageParsedTag, err := name.NewTag(tag, name.WithDefaultRegistry(""))
+		if err != nil {
+			return nil
```

Comment:
you shouldn't just swallow the error here. you should return it and bubble it up. in fact, it would catch the scenario where i put in a bad tag like:
```
acorn tag d53545e106fb 'foo:a@asdfasdfadsf'
```

With your current code, that would just silently drop the tag and return no useful feedback to the user. if you actually return the error, the user will get a useful message

Reply:
can you please add tests cases to assert this fails